src/test/basics/RangeSet_test.cpp

```diff
@@ -26,69 +26,175 @@ namespace ripple {
-    {
-        testcase ("membership");
+        // Adding 5 creates complete range
+        r2.setValue(5);
+        for (int i = 1; i <= 21; ++i)
+        {
+            BEAST_EXPECT(r2.hasValue(i));
+        }
 
-        RangeSet r1, r2;
+        // Additional tests to complete coverage
+
+        // Isolate 1
+        r2.clearValue(2);
+        for (int i = 1; i <= 21; ++i)
+        {
+            if (i != 2)
+                BEAST_EXPECT(r2.hasValue(i));
+            else
+                BEAST_EXPECT(!r2.hasValue(i));
+        }
+        // Remove 1 as well and shrink 21
+        r2.clearValue(1);
+        r2.clearValue(21);
+        for (int i = 1; i <= 21; ++i)
+        {
+            if (i > 2 && i < 21)
+                BEAST_EXPECT(r2.hasValue(i));
+            else
+                BEAST_EXPECT(!r2.hasValue(i));
+        }
 
-        r1.setRange (1, 10);
-        r1.clearValue (5);
-        r1.setRange (11, 20);
 
-        r2.setRange (1, 4);
-        r2.setRange (6, 10);
-        r2.setRange (10, 20);
+    }
 
-        BEAST_EXPECT(!r1.hasValue (5));
+    void
+    testOverlappingRanges()
+    {
+        RangeSet r;
 
-        BEAST_EXPECT(r2.hasValue (9));
+        r.setRange(4,9);
+        for (int i = 1; i <= 10; ++i)
+        {
+            if (i >=4 && i <= 9)
+                BEAST_EXPECT(r.hasValue(i));
+            else
+                BEAST_EXPECT(!r.hasValue(i));
+        }
+
+        // Add overlap range with front end 
+        r.setRange(1,5);
+        for (int i = 1; i <= 10; ++i)
+        {
+            if (i <= 9)
+                BEAST_EXPECT(r.hasValue(i));
+            else
+                BEAST_EXPECT(!r.hasValue(i));
+        }
+        
+        // Add overlap with end
+        r.setRange(7,10);
+        for (int i = 1; i <= 10; ++i)
+        {
+            if(!BEAST_EXPECT(r.hasValue(i)))
+                std::cout << i << "\n";
+        }
+
+        // Add existing range in middle
+        r.setRange(5,7);
+        for (int i = 1; i <= 10; ++i)
+        {
+            BEAST_EXPECT(r.hasValue(i));
+        }
     }
 
-    void testPrevMissing ()
+    void
+    testPrevMissing()
     {
-        testcase ("prevMissing");
+        testcase("prevMissing");
+
+        // Set will include:
+        // [ 0, 5]
+        // [10,15]
+        // [20,25]
+        // etc...
 
-        RangeSet const set = createPredefinedSet ();
+        RangeSet set;
+        for (int i = 0; i < 10; ++i)
+            set.setRange(10 * i, 10 * i + 5);
 
         for (int i = 0; i < 100; ++i)
         {
-            int const oneBelowRange = (10*(i/10))-1;
+            int const oneBelowRange = (10 * (i / 10)) - 1;
 
             int const expectedPrevMissing =
-                ((i % 10) > 6) ? (i-1) : oneBelowRange;
+                ((i % 10) > 6) ? (i - 1) : oneBelowRange;
 
-            BEAST_EXPECT(set.prevMissing (i) == expectedPrevMissing);
+            BEAST_EXPECT(set.prevMissing(i) == expectedPrevMissing);
```

Comment:
So, `prevMissing` returns a `std::uint32_t` (which may be initialized to `static_int<std::uint32_t>(-1)`) and `expectedPrevMissing` is an `int`. This smells.